HACKING.md

```diff
@@ -203,25 +203,29 @@ Coding Style
   *unless they avoid great repetition*. On the average case,
   just put the assignment on the previous line. So try NOT to do
   the following:
-  ~~```if ((ret = open(...)) == -1)```~~
+  ~~`if ((ret = open(...)) == -1)`~~
 * Explicit comparisons are better than implicit, i.e. prefer writing
-  ```if (number == 0)``` or ```if (pointer == NULL)```
-  instead of ~~```if (!number)```~~ or ~~```if (!pointer)```~~. It only makes
-  sense to test booleans directly, for example ```if (is_valid)``` is good.
+  `if (number == 0)` or `if (pointer == NULL)`
+  instead of ~~`if (!number)`~~ or ~~`if (!pointer)`~~. It only makes
+  sense to test booleans directly, for example `if (is_valid)` is good.
   Furthermore have the literal last in the comparison, not first, i.e.
-  prefer writing ```if (open(...) == -1)``` instead of
-  ~~```if (-1 == open(...))```~~.
+  prefer writing `if (open(...) == -1)` instead of
+  ~~`if (-1 == open(...))`~~.
 * Control statements need to have braces on separate line,
   no matter how simple they are.
   * Caution, do-while loops should have the closing brace at the same
     line with while, so that it can't be confused with empty while statement.
     ```c
     do
     {
-        /* ... */
+        // ...
 
     } while (condition);
     ```
+* In functions which can fail, error code (int) or success/failure (bool) should be returned.
+    * `true`(bool) and `0`(int) should always signify success.
```

Comment:
zero for success in int functions might be a bit dangerous, because zero is `false`. In the end it's just a convention so I'm OK, I'm not sure if I would use it personally.

Reply:
I agree, but `success = 0` seems the most standard. It is already used many places in our codebase, and fits in nicely with exit codes, where success is also 0.